### PR TITLE
Bookmarks

### PR DESCRIPTION
**Pull request by nxrara**
- edited schema to add bookmarks
- edited routes to add bookmarks
- added controller, index.html.erb for bookmarks
- next branch to change bookmark link to button ajax

### CODE REVIEW

File: app/controllers/bookmarks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    if @bookmark.save
+      redirect_to bookmarks_path, notice: "Trip is saved as bookmark!"
+    else
+      redirect_to trail_path(@trail), :alert => @bookmark.errors.full_messages[0]
+    end
+  end
+
+  def destroy
+    @bookmark.destroy
```
Comment (barbayjuliette) [inline app/controllers/bookmarks_controller.rb:19]: You have to define @bookmark first
Screenshot 2022-11-24 at 16 59 15

File: app/models/user.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class User < ApplicationRecord
   has_many :trips, dependent: :destroy
+  has_many :bookmarks, dependent: :destroy
```
Comment (barbayjuliette) [inline app/models/user.rb:3]: A user has many bookmarks 
AND a trail has many booksmarks

File: app/views/bookmarks/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<h1>Your Bookmarks</h1>
```
Comment (barbayjuliette) [inline app/views/bookmarks/index.html.erb:1]: WIll have to be adapted to the new cards. 
I could put my cards in a partial ?

File: config/routes.rb [hunk 4]
```diff
@@ -8,9 +8,11 @@
```
Comment (barbayjuliette) [inline config/routes.rb:8]: I am not sure about the routes.
@lxingern what do you think?
WHy is create nested and destroy is not?

File: test/controllers/bookmarks_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (barbayjuliette) [inline test/controllers/bookmarks_controller_test.rb:1]: What is this file ?

File: test/models/bookmark_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (barbayjuliette) [inline test/models/bookmark_test.rb:1]: What is this file ?

**Review by barbayjuliette — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 4 above)
Comment (nxrara) [reply config/routes.rb:8]: verified that destroy is not to be nested. same for create

File: app/views/bookmarks/index.html.erb (hunk 3 above)
Comment (nxrara) [reply app/views/bookmarks/index.html.erb]: yes, left <%= render for your partial once your branch is merged

**Pull request merged**